packages/markers/src/browser/problem/problem-tabbar-decorator.ts

```diff
@@ -0,0 +1,205 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable, postConstruct } from 'inversify';
+import { Diagnostic, DiagnosticSeverity } from 'vscode-languageserver-types';
+import URI from '@theia/core/lib/common/uri';
+import { notEmpty } from '@theia/core/lib/common/objects';
+import { Event, Emitter } from '@theia/core/lib/common/event';
+import { Title, Widget } from '@phosphor/widgets';
+import { WidgetDecoration } from '@theia/core/lib/browser/widget-decoration';
+import { TabBarDecorator } from '@theia/core/lib/browser/shell/tab-bar-decorator';
+import { Marker } from '../../common/marker';
+import { ProblemManager } from './problem-manager';
+import { WorkspaceService } from '@theia/workspace/lib/browser';
+import { ProblemPrefenreces, ProblemConfiguration } from './problem-preferences';
+import { PreferenceChangeEvent } from '@theia/core/lib/browser';
+@injectable()
+export class ProblemTabBarDecorator implements TabBarDecorator {
+
+    readonly id = 'theia-problem-tabbar-decorator';
+
+    protected emitter: Emitter<void>;
+
+    protected enabled: boolean;
```

Comment:
please just use `this.preferences['problem.decorations.tabbar.enabled']`, don't duplicate state, you have to sync it then and cause of many memory leaks and bugs

Reply:
Sure. Just updated it!